### ISSUE TITLE
PdfMetadata#load_from_docinfo() fails with "TypeError: ['']"

### ISSUE THREAD

**Issue by kbengs** [pikepdf/pikepdf#162]
Code:
```
import pikepdf

print(pikepdf.__version__)

doc = pikepdf.open('/home/kb/tmp/test/testpdf.pdf')
with doc.open_metadata() as meta:
    meta.load_from_docinfo(doc.docinfo)
```

Output:
```
2.3.0
Traceback (most recent call last):
  File "./testpy.py", line 7, in <module>
    meta.load_from_docinfo(doc.docinfo)
  File "/home/kb/.local/lib/python3.8/site-packages/pikepdf/models/metadata.py", line 382, in load_from_docinfo
    self._setitem(qname, val, True)
  File "/home/kb/.local/lib/python3.8/site-packages/pikepdf/models/metadata.py", line 725, in _setitem
    raise TypeError(val)
TypeError: ['']
```

Unfortunately I can't share the file

The issue is reported here pdfarranger/pdfarranger#384

**Comment by jbarlow83** [comment]
I am guessing that the XMP metadata in this PDF has a statement like:

```xml
<rdf:Description rdf:about="" xmlns:dc="http://purl.org/dc/elements/1.1/" dc:creator="Foo"></rdf:Description>
```

The trouble is that `dc:creator` is supposed to be an ordered list of strings, but it's been created as an attribute. Because of that we have delete the whole tag and reconstruct it.

Fixed in v2.4.0

**Comment by kbengs** [comment]
The output in v2.4.0 is: 
```
UserWarning: Merging elements of ['']
  warn(f"Merging elements of {s}")
```
I guess that is right. Thank you for the fix.
